### PR TITLE
Fix #96: Return 404 for invalid chart IDs

### LINKED ISSUE #96: Return a 404 when a metric of a given ID (e.g. /chart/999/) can't be found
### Steps to reproduce

Visit the chart detail page of a chart that doesn't exist (e.g. https://moz-distribution-viewer.herokuapp.com/chart/999/)
### Actual result

An error is printed to the console and the spinners spin indefinitely
### Expected result

404

### PR DESCRIPTION
**Pull request by openjck**

### CODE REVIEW

**Comment by robhudson** [comment]
This has conflicts now.

**Comment by openjck** [comment]
Updated

**Comment by openjck** [comment]
Let's merge this after #111 so as not to cause another rebase for it.

File: distributionviewer/core/static/js/app/components/containers/chart-detail-container.js [hunk 1]
```diff
@@ -16,20 +17,25 @@ class ChartDetailContainer extends React.Component {
...
+      );
+    }
   }
 }
 
 const mapStateToProps = function(store) {
   return {
-    metric: store.metricState.metric
+    metric: store.metricState.metric,
+    status: store.metricState.status,
```
Comment (robhudson) [inline distributionviewer/core/static/js/app/components/containers/chart-detail-container.js:38]: Should I quell the urge to complain about little stuff like this? Babel transpiles it all away anyway, right?

**Comment by openjck** [comment]
No merge conflicts, so this is all set

File: distributionviewer/core/static/js/app/components/containers/chart-detail-container.js (hunk 1 above)
Comment (openjck) [reply distributionviewer/core/static/js/app/components/containers/chart-detail-container.js:38]: I'm happy either way. Trailing commas are valid as of ES5, simplify adding and removing elements, and make diffs more readable. But they do cause errors in IE8 and below.

Here are [some arguments](http://eslint.org/docs/rules/comma-dangle) for and against trailing commas.

Comment (openjck) [reply distributionviewer/core/static/js/app/components/containers/chart-detail-container.js:38]: And you're right, it looks like Babel does [remove trialing commas](https://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-2&code=var%20obj%20%3D%20%7B%0A%20%20a%3A%201%2C%0A%20%20b%3A%202%2C%0A%7D%3B) to play it safe.

Comment (robhudson) [reply distributionviewer/core/static/js/app/components/containers/chart-detail-container.js:38]: I wasn't expecting such a detailed reply but thanks. Interesting to know and see the babel conversion live. That's cool.

**Pull request merged**